Open Judge System/Services/OJS.Services/Business/Participants/ParticipantsBusinessService.cs

```diff
@@ -146,6 +152,26 @@ public void UpdateTotalScoreSnapshotOfParticipants()
             this.participantsData.UpdateTotalScoreSnapshot();
         }
 
+        public void RemoveParticipantMultipleScores()
+        {
+            var participants = 
+                this.scoresDataService.GetAll()
+                .GroupBy(ps => new { ps.IsOfficial, ps.ProblemId, ps.ParticipantId })
+                .Where(ps=>ps.Count() > 1)
+                .ToList();
+            
+            using (var scope = TransactionsHelper.CreateTransactionScope())
+            {
+                foreach (var participant in participants)
+                {
+                    var participantsToRemove = participant.OrderByDescending(ps => ps.Points).Skip(1).ToList();
+                    this.scoresDataService.Delete(participantsToRemove);
+
+                }
```

Comment:
Why this is in Transaction scope?

Reply:
Because delete is calling SaveChanges and i think is better idea to make only one transaction for all participantScores instead of many.

Reply:
You can achieve this without transaction scope

Reply:
Done